python/ql/src/experimental/Security/CWE-340/TokenBuiltFromUUID.ql

```diff
@@ -0,0 +1,69 @@
+/**
+ * @name Predictable token
+ * @description Tokens used for sensitive tasks, such as, password recovery,
+ *  and email confirmation, should not use predictable values.
+ * @kind path-problem
+ * @precision medium
+ * @problem.severity error
+ * @security-severity 5
+ * @id py/predictable-token
+ * @tags security
+ *       external/cwe/cwe-340
+ */
+
+import python
+import semmle.python.dataflow.new.DataFlow
+import semmle.python.ApiGraphs
+import semmle.python.dataflow.new.TaintTracking
+import DataFlow::PathGraph
+
+class PredictableResultSource extends DataFlow::Node {
+  PredictableResultSource() {
+    exists(API::Node uuidCallRet |
+      uuidCallRet =
+        API::moduleImport("uuid")
+            .getMember(["uuid1", "uuid2", "uuid3", "uuid5"])
+            .getACall()
+            .getReturn()
```

Comment:
```suggestion
            .getReturn()
```
You should be able to replace this with just `getReturn`.

Reply:
Thanks for the suggestion. This was my first try with the python CodeQL API, so there was some trial and error involved. =)